onnxruntime/core/providers/common.h

```diff
@@ -42,4 +42,79 @@ inline float clamp(float v, float lo, float hi) {
   return v;
 }
 
+enum class AutoPadType {
+  NOTSET = 0,
+  VALID = 1,
+  SAME_UPPER = 2,
+  SAME_LOWER = 3,
+};
+
+inline AutoPadType StringToAutoPadType(const std::string& str) {
+  if (str.empty()) {
+    return AutoPadType::NOTSET;
+  }
+  if (str == "NOTSET") {  // in onnx spec, default value is "NOTSET"
+    return AutoPadType::NOTSET;
+  }
+  if (str == "VALID") {
+    return AutoPadType::VALID;
+  }
+  if (str == "SAME_UPPER") {
+    return AutoPadType::SAME_UPPER;
+  }
+  if (str == "SAME_LOWER") {
+    return AutoPadType::SAME_LOWER;
+  }
+  ORT_ENFORCE(false, "Unknown AutoPadType String");
+}
+
+// helper function
+template <bool ForceSymmetricAutoPadding>
+Status ComputePadAndOutputShape(
```

Comment:
>ComputePadAndOutputShape [](start = 7, length = 24)

Is it possible to split this into 2 separated functions? ComputePad and ComputeOutputShape
For some EP, NNAPI to be specific, we only need to ComputePad once, but may call ComputeOutputShape multiple times and prefer pad_head and pad_tail to be const

Reply:
That sounds like a reasonable idea! I'm going to go ahead and merge this change as is to keep it small, but I'd support the idea of splitting the functions in the future.